packages-next/fields/src/types/decimal/index.ts

```diff
@@ -23,6 +23,6 @@ export const decimal = <TGeneratedListTypes extends BaseGeneratedListTypes>(
   views: resolveView('decimal/views'),
   getAdminMeta: () => ({
     precision: config.precision || null,
-    scale: config.scale || null,
+    scale: config.scale == null ? null : config.scale,
```

Comment:
do you want to check if data type is integer rather than null?

Reply:
Yeah I'd flip this around, check that `config.scale` is an integer or otherwise set it to null

Reply:
Yeah, I agree

Reply:
Yes, I fixed this.

Reply:
Hmm, trying to add `isInteger` check produces the typescript error:
```js
    scale: Number.isInteger(config.scale) ? config.scale : null,
```
```
$ tsc
Error: packages-next/fields/src/types/decimal/index.ts(24,3): error TS2322: Type '() => { precision: number | null; scale: number | null | undefined; }' is not assignable to type '(listKey: string, path: string, adminMeta: AdminMetaRootVal) => JSONValue'.
  Type '{ precision: number | null; scale: number | null | undefined; }' is not assignable to type 'JSONValue'.
    Type '{ precision: number | null; scale: number | null | undefined; }' is not assignable to type '{ [key: string]: JSONValue; }'.
      Property 'scale' is incompatible with index signature.
        Type 'number | null | undefined' is not assignable to type 'JSONValue'.
          Type 'undefined' is not assignable to type 'JSONValue'.

Reply:
Maybe simple checking like this will be enough?
```
    scale: config.scale != null ? config.scale : null,
```
I committed this variant now, but glad to see better fix.